sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -313,6 +313,22 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
     createTable(tableName, source, new StructType, options)
   }
 
+  /**
+   * (Scala-specific)
+   * Creates a table based on the dataset in a data source and a set of options.
+   * Then, returns the corresponding DataFrame.
+   *
+   * @group ddl_ops
+   * @since 3.1.0
+   */
+  override def createTable(
+      tableName: String,
+      source: String,
+      options: Map[String, String],
+      description: Option[String]): DataFrame = {
```

Comment:
Oh yeah, we shouldn't use `Option` because this is a Scala specific collection. We will just have to override so Java side can call these as well.

Reply:
I've switched to a plain `String` in the Java version of this API. (Not sure if I'm using the terminology correctly.)